src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -500,3 +503,102 @@ export async function refundRequestIsValid(
 
   return { valid: true };
 }
+
+export type SpokePoolEventFilter = {
+  originChainId?: number;
+  destinationChainId?: number;
+  relayer?: string;
+  fromBlock?: number;
+};
+
+export type SpokePoolFillFilter = SpokePoolEventFilter & {
+  repaymentChainId?: number;
+  isSlowRelay?: boolean;
+};
+
+// @description Search for fills recorded by a specific SpokePool.
+// @param chainId Chain ID of the relevant SpokePoolClient instance.
+// @param spokePoolClients Set of SpokePoolClient instances, mapped by chain ID.
+// @param filter  Optional filtering criteria.
+// @returns Array of FillWithBlock events matching the chain ID and optional filtering criteria.
+export async function getFills(
+  chainId: number,
+  spokePoolClients: SpokePoolClients,
+  filter: SpokePoolFillFilter = {}
+): Promise<FillWithBlock[]> {
+  const spokePoolClient = spokePoolClients[chainId];
+  assert(isDefined(spokePoolClient));
+
+  const { originChainId, repaymentChainId, relayer, isSlowRelay, fromBlock } = filter;
+
+  const fills = await filterAsync(spokePoolClient.getFills(), async (fill) => {
+    if (isDefined(fromBlock) && fill.blockNumber > fromBlock) {
```

Comment:
```suggestion
    if (!isDefined(fromBlock) || fill.blockNumber < fromBlock) {
```

Reply:
or should it be
```suggestion
    if (isDefined(fromBlock) && fill.blockNumber < fromBlock) {
```

Reply:
Yeah the latter is correct. If `fromBlock` is undefined then we should not filter on `blockNumber`. Has been fixed.